src/bls12_381/fq2.rs

```diff
@@ -145,6 +156,11 @@ impl Field for Fq2 {
 }
 
 impl SqrtField for Fq2 {
+
+    fn legendre(&self) -> ::LegendreSymbol {
+        Fq2::norm(&self).legendre()
+    }
+
     fn sqrt(&self) -> Option<Self> {
         // Algorithm 9, https://eprint.iacr.org/2012/685.pdf
 
```

Comment:
Not related to this PR, but while I've noticed it: move the magic constants for (q - 3)/4 and (q - 1)/2 into the trait initialisation and assert they are correct if needed.